pkg/plugin/plugin_test.go

```diff
@@ -337,6 +338,76 @@ var _ = Describe("Plugin - test CNI command flows", func() {
 				Expect(updatedPluginConf.MAC).To(BeIdenticalTo(testValidMAC3))
 			})
 		})
+
+		Context("Update DeviceInfo", func() {
+			var (
+				tmpFile string
+			)
+
+			BeforeEach(func() {
+				f, err := os.CreateTemp("", "accbr-deviceinfo")
+				Expect(err).NotTo(HaveOccurred())
+				tmpFile = f.Name()
+			})
+
+			AfterEach(func() {
+				Expect(os.Remove(tmpFile)).NotTo(HaveOccurred())
+			})
+
+			It("succeed", func() {
+				cmdCtx := &cmdContext{pluginConf: &localtypes.PluginConf{}}
+				cmdCtx.pluginConf.Representor = "eth3"
+				cmdCtx.pluginConf.RuntimeConfig.CNIDeviceInfoFile = tmpFile
+
+				By("DeviceInfo obj is empty")
+				Expect(os.WriteFile(tmpFile, []byte(`{}`), 0600)).NotTo(HaveOccurred())
+				Expect(plugin.updateDeviceInfo(cmdCtx)).NotTo(HaveOccurred())
+				result, err := os.ReadFile(tmpFile)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(result).To(MatchJSON(`{"pci": {"vf-representor-device": "eth3"}}`))
+
+				By("merge DeviceInfo")
+				Expect(os.WriteFile(tmpFile,
+					[]byte(`{"pci": {"pci-address": "0000:d8:00.2"}}`), 0600)).NotTo(HaveOccurred())
+				Expect(plugin.updateDeviceInfo(cmdCtx)).NotTo(HaveOccurred())
+				result, err = os.ReadFile(tmpFile)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(result).To(MatchJSON(
+					`{"pci": {"pci-address": "0000:d8:00.2", "vf-representor-device": "eth3"}}`))
```

Comment:
should "vf-representor-device" be describe in [1]?
Also why not expose also  the pf-pci-address?  Don't you have it? 

[1] - https://github.com/k8snetworkplumbingwg/device-info-spec/blob/main/SPEC.md

Reply:
> should "vf-representor-device" be describe in [1]?

Yes, I already created PR to spec repo https://github.com/k8snetworkplumbingwg/device-info-spec/pull/3

Also, Multus will not report "vf-representor-device" field in `k8s.v1.cni.cncf.io/network-status` annotation before we update [network-attachment-definition-client](https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/blob/6ace5615d26a4d00d0a24bbbfa5d5f400ce7593e/pkg/apis/k8s.cni.cncf.io/v1/types.go#L60) dependency.

> Also why not expose also the pf-pci-address?

I will check what else we can report

Reply:
It is quite easy to retrieve `pf-pci-address`. I think this information can be useful, so I update PR to also expose `pf-pci-address` field

Reply:
> Also why not expose also the pf-pci-address? Don't you have it?

This should be exposed by sriov-network-device-plugin IMO where applicable (e.g not in VM use-case)

Reply:
and how do you see it in the network-status?

Reply:
if we extend device plugin to add it:

https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/blob/master/pkg/netdevice/netResourcePool.go#L98

then, according to device info spec section 4.2.1 , it copies the device plugin device info file to cni path,
then CNI can use it. in our case it can mean just read it, add VF representor and write it back.
then multus exposes it via annotations.

Reply:
so you mean the SR-IOV device plugin will have the logic for the VF representor  lookup and all cni just need to read it and plug it to the bridge?
maybe we need to support both places if someone will work with old SR-IOV device plugin

Reply:
no, i just mean that `pf-pci-address` should be added to device info file by device plugin and not cni.

Reply:
i dont think device plugin should be aware of it

Reply:
should be aware of what? That the VF is switchdev? how is different from vdpa?

Reply:
on problem is that VF repo is renameable, but I see the spec allows also rdma device which is also renamable. 
Is any one uses rdma-device?

Reply:
> should be aware of what? That the VF is switchdev? how is different from vdpa?

well, it depends where do you draw the line right ? almost everything we handle is some sort of device...
one example is that you explicitly ask for VDPA in the resource pool. you do not do it today for switchdev.

also with VDPA, you can assign directly to container without CNI for DPDK use case IIRC.
so there would be no CNI to add this information.

as you say VF rep is rename-able. so perhaps it should be CNI (the entity who might do the renaming) who adds it.
moreover, there are CNIs who use VF but do not care about the representor. so it should not be added to network status annot in this case.

as for RDMA devices: it is defined in device-info-spec but is not set by any entity that im aware and IMO is a separate discussion.

IMO this PR should be limited to adding just `representor-device` attribute.
`pf-pci-address` if desirable can be set in device-plugin.